sphinx_gallery/utils.py

```diff
@@ -126,8 +126,8 @@ def replace_py_ipynb(fname):
 
 def get_md5sum(src_file):
     """Returns md5sum of file"""
-    with open(src_file, 'rb') as src_data:
-        src_content = src_data.read()
+    with open(src_file, 'rt', errors='surrogateescape') as src_data:
```

Comment:
Some of the things we md5sum are binary pickle objects. It seems like it would be better to add a `get_md5sum(src_file, mode='b')` default, and if `mode='t'` then you use this escape-replacement version, and the default binary mode stays the same (plain `'rb'` read).

Reply:
How about checking _src_file_ extension is ".py"?

Reply:
There are other things we check, too, like `ipynb` that can be text mode. I'd prefer to keep the decision about which mode to use with the places we make calls to `_replace_md5` (which calls `get_md5sum` internally) or `get_md5sum` directly